test/e2e/fishapp/dynamic_stack_test.go

```diff
@@ -0,0 +1,111 @@
+package fishapp_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/fishapp"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/fishapp/shared"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/framework"
+	. "github.com/onsi/ginkgo"
+	"time"
+)
+
+var _ = Describe("tests by dynamic generated stack", func() {
+	var (
+		ctx context.Context
+		f   *framework.Framework
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		f = framework.New(framework.GlobalOptions)
+
+		if f.Options.ControllerImage != "" {
+			By("reset cluster with default controller", func() {
+				f.HelmManager.ResetAppMeshController()
+			})
+		}
+		if f.Options.InjectorImage != "" {
+			By("reset cluster with default injector", func() {
+				f.HelmManager.ResetAppMeshInjector()
+			})
+		}
+	})
+
+	Context("normal test dimensions", func() {
+		var stackPrototype fishapp.DynamicStack
+		var stacksPendingCleanUp []*fishapp.DynamicStack
+
+		BeforeEach(func() {
+			stackPrototype = fishapp.DynamicStack{
+				VirtualServicesCount:        5,
+				VirtualNodesCount:           5,
+				RoutesCountPerVirtualRouter: 2,
+				TargetsCountPerRoute:        4,
+				BackendsCountPerVirtualNode: 2,
+				ReplicasPerVirtualNode:      3,
+				ConnectivityCheckPerURL:     400,
+			}
+			stacksPendingCleanUp = nil
+		})
+
+		AfterEach(func() {
+			for _, stack := range stacksPendingCleanUp {
+				stack.Cleanup(ctx, f)
+			}
+		})
+
+		for _, sdType := range []shared.ServiceDiscoveryType{shared.DNSServiceDiscovery, shared.CloudMapServiceDiscovery} {
+			func(sdType shared.ServiceDiscoveryType) {
+				It(fmt.Sprintf("should behaves correctly with service discovery type %v", sdType), func() {
```

Comment:
"should behave correctly" is a bit vague, can you 1. split this into multiple tests and 2. improve the description to something like "should helm install and upgrade for controller and injector".

Reply:
@nckturner what do you mean by split this into multiple tests. 
There are two test currently DNS and CloudMap. and within the test unit, the tests are sequential, so not really split able.
`install old controller`  => `deploy stack` => `verify behavior` => 
`install new controller` => `verify behavior against old stack` =>
`deploy new stack` => `verify behavior against new stack`

Also, for the behavior, the test is just check connectivity and result distribution.